components/script/dom/mutationobserver.rs

```diff
@@ -0,0 +1,40 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use core::ptr::null;
+use dom;
+use dom::bindings::codegen::Bindings::MutationObserverBinding::MutationCallback;
+use dom::bindings::codegen::Bindings::MutationObserverBinding::MutationObserverInit;
+use dom::bindings::error::Fallible;
+use dom::bindings::js::Root;
+use dom::bindings::reflector::Reflector;
+use dom::bindings::trace::JSTraceable;
+use dom::mutationrecord::MutationRecord;
+use dom::node::Node;
+use dom::window::Window;
+use dom_struct::dom_struct;
+use script_thread::ScriptThread;
+use std::ops::Deref;
+use std::rc::Rc;
+
+#[dom_struct]
+pub struct MutationObserver {
+    reflector_: Reflector,
+    callback: MutationCallback,
```

Comment:
Thanks Josh,

I have fixed the test-tidy issues, for some reason those were not visible when I ran the command locally.
I have also made the other changes as per the comments. Now the mutation observer part is not giving any errors, but there is an error in the script thread file at line 583:
"error: the type of this value must be known in this context" I cannot figure out why is it an error, the method is exactly the same as the mark_document_with_no_blocked_loads method.